### ISSUE TITLE
Linux build error (0.2.8.1)

### ISSUE THREAD

**Issue by 3laws** [sorayuki/obs-multi-rtmp#220]
Similar to #219, I just renamed my `libobsConfig.cmake` to `LibObsConfig.cmake`.

However, now I get the following:

<details> 
<summary><b>INSTALL=1 LibObs_DIR=/home/3laws/.src/obs-studio/build/libobs ./build_linux.sh</b></summary> 

```cmake,
+ : 1
+ : /usr
+ : ''
+ : /usr/include/obs
++ grep 'obs-multi-rtmp VERSION' CMakeLists.txt
+ ver='project(obs-multi-rtmp VERSION 0.2.8.0)'
+ ver=($ver)
+ ver='0.2.8.0)'
+ ver=0.2.8.0
+ rm -fr build dist '*.tar.xz'
+ cmake -DCMAKE_INSTALL_PREFIX=/usr -DQTDIR= -DOBS_SRC_DIR=/usr/include/obs -B build .
-- The C compiler identification is GNU 12.2.1
-- The CXX compiler identification is GNU 12.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib64/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE  
-- Found OpenGL: /usr/lib64/libOpenGL.so   
-- Found WrapOpenGL: TRUE  
-- Found XKB: /usr/lib64/libxkbcommon.so (found suitable version "1.4.1", minimum required is "0.5.0") 
-- Found WrapVulkanHeaders: /usr/include  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/3laws/.src/obs-multi-rtmp/build
+ cmake --build build --config Release
[ 20%] Building CXX object CMakeFiles/obs-multi-rtmp.dir/cmake_pch.hxx.gch
In file included from /home/3laws/.src/obs-multi-rtmp/build/CMakeFiles/obs-multi-rtmp.dir/cmake_pch.hxx:5,
                 from <command-line>:
/home/3laws/.src/obs-multi-rtmp/src/pch.h:30:10: fatal error: obs-module.h: No such file or directory
   30 | #include "obs-module.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [CMakeFiles/obs-multi-rtmp.dir/build.make:77: CMakeFiles/obs-multi-rtmp.dir/cmake_pch.hxx.gch] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/obs-multi-rtmp.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```
</details>

EDIT: Completely forgot about distro info. Fedora 37.

**Comment by 3laws** [comment]
Did some more config of the files, now stuck in here:

<details>
<summary><b> INSTALL=1 LibObs_DIR=/home/3laws/.src/obs-studio/build/libobs ./build_linux.sh
</b></summary> 

```cmake,
+ cmake --build build --config Release
[ 20%] Building CXX object CMakeFiles/obs-multi-rtmp.dir/cmake_pch.hxx.gch
[ 40%] Building CXX object CMakeFiles/obs-multi-rtmp.dir/src/obs-multi-rtmp.cpp.o
[ 60%] Building CXX object CMakeFiles/obs-multi-rtmp.dir/src/edit-widget.cpp.o
[ 80%] Building CXX object CMakeFiles/obs-multi-rtmp.dir/src/push-widget.cpp.o
/home/3laws/Software/obs-multi-rtmp-0.2.8.1-OBS28/src/push-widget.cpp: In member function ‘bool PushWidgetImpl::PrepareOutputEncoders()’:
/home/3laws/Software/obs-multi-rtmp-0.2.8.1-OBS28/src/push-widget.cpp:243:35: warning: ‘void obs_encoder_addref(obs_encoder_t*)’ is deprecated [-Wdeprecated-declarations]
  243 |                 obs_encoder_addref(venc);
      |                 ~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /home/3laws/.src/obs-studio/libobs/obs-module.h:20,
                 from /home/3laws/Software/obs-multi-rtmp-0.2.8.1-OBS28/src/pch.h:30,
                 from /home/3laws/Software/obs-multi-rtmp-0.2.8.1-OBS28/build/CMakeFiles/obs-multi-rtmp.dir/cmake_pch.hxx:5,
                 from <command-line>:
/home/3laws/.src/obs-studio/libobs/obs.h:2274:37: note: declared here
 2274 | OBS_EXTERNAL_DEPRECATED EXPORT void obs_encoder_addref(obs_encoder_t *encoder);
      |                                     ^~~~~~~~~~~~~~~~~~
/home/3laws/Software/obs-multi-rtmp-0.2.8.1-OBS28/src/push-widget.cpp:248:35: warning: ‘void obs_encoder_addref(obs_encoder_t*)’ is deprecated [-Wdeprecated-declarations]
  248 |                 obs_encoder_addref(aenc);
      |                 ~~~~~~~~~~~~~~~~~~^~~~~~
/home/3laws/.src/obs-studio/libobs/obs.h:2274:37: note: declared here
 2274 | OBS_EXTERNAL_DEPRECATED EXPORT void obs_encoder_addref(obs_encoder_t *encoder);
      |                                     ^~~~~~~~~~~~~~~~~~
[100%] Linking CXX shared library obs-multi-rtmp.so
/usr/bin/ld: cannot find -llibobs: No such file or directory
collect2: error: ld returned 1 exit status
gmake[2]: *** [CMakeFiles/obs-multi-rtmp.dir/build.make:156: obs-multi-rtmp.so] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/obs-multi-rtmp.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```
</details>

**Comment by sorayuki** [comment]
the makefile is written for windows.
it may not has the name "libobs" on linux.

would you like to change 
```set(LIB_OBS libobs)```
to
```set(LIB_OBS obs)```
and have another try?

**Comment by Ryu945** [comment]
Keep in mind it will only build correctly if you have an exact copy of the project.   Downloading the source and compiling will not work.   Everything has to be located correctly.  I found this out on Fedora 36.

https://github.com/sorayuki/obs-multi-rtmp/archive/refs/heads/master.zip

I can upload my Fedora 36 file I compiled though I am not sure how to do that to Github.

To get the dependencies I needed to compile this project, I installed the following programs on Fedora 36.

```
sudo dnf install gcc-c++
sudo dnf install qt6-qtbase-devel
```

I don't know how these requirements may change on Fedora 37.
